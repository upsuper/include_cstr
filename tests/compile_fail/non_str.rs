use include_cstr::include_cstr;

fn main() {
    let _ = include_cstr!(1);
    let _ = include_cstr!(&1);
    let _ = include_cstr!(true);
    let _ = include_cstr!(("a"));
    let _ = include_cstr!('a');
    let _ = include_cstr!(b'a');
    let _ = include_cstr!(b"a");
}
