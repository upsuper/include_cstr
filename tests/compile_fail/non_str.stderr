error: expected a string literal
 --> $DIR/non_str.rs:4:27
  |
4 |     let _ = include_cstr!(1);
  |                           ^

error: expected a string literal
 --> $DIR/non_str.rs:5:27
  |
5 |     let _ = include_cstr!(&1);
  |                           ^

error: expected a string literal
 --> $DIR/non_str.rs:6:27
  |
6 |     let _ = include_cstr!(true);
  |                           ^^^^

error: expected a string literal
 --> $DIR/non_str.rs:7:27
  |
7 |     let _ = include_cstr!(("a"));
  |                           ^^^^^

error: expected a string literal
 --> $DIR/non_str.rs:8:27
  |
8 |     let _ = include_cstr!('a');
  |                           ^^^

error: expected a string literal
 --> $DIR/non_str.rs:9:27
  |
9 |     let _ = include_cstr!(b'a');
  |                           ^^^^

error: expected a string literal
  --> $DIR/non_str.rs:10:27
   |
10 |     let _ = include_cstr!(b"a");
   |                           ^^^^
